### ISSUE TITLE
async/sync maxDepth broken

### ISSUE THREAD

**Issue by borestad** [thecodrr/fdir#12]
```
const fdir = require('fdir')

fdir.async('.', { maxDepth: 99 }).then((res) => {
  console.log(res, res.length)
})

```

In a rather large folder structure, the async & sync method with maxDepth doesn't work properly:

```
[
  '/Users/johan/test/test-repos/.DS_Store',
  '/Users/johan/test/test-repos/.gitignore',
  '/Users/johan/test/test-repos/.jscpd.json',
  '/Users/johan/test/test-repos/.mrconfig',
  '/Users/johan/test/test-repos/Jakefile',
  '/Users/johan/test/test-repos/Makefile',
  '/Users/johan/test/test-repos/README.md',
  '/Users/johan/test/test-repos/async.js',
  '/Users/johan/test/test-repos/distribution.txt',
  '/Users/johan/test/test-repos/gamerules.txt',
  '/Users/johan/test/test-repos/games.txt',
  '/Users/johan/test/test-repos/gfw.txt',
  '/Users/johan/test/test-repos/langlib.txt',
  '/Users/johan/test/test-repos/package.json',
  '/Users/johan/test/test-repos/project.txt',
  '/Users/johan/test/test-repos/stats.txt',
  '/Users/johan/test/test-repos/yarn.lock',
  '/Users/johan/test/test-repos/.git/COMMIT_EDITMSG',
  '/Users/johan/test/test-repos/.git/FETCH_HEAD',
  '/Users/johan/test/test-repos/.git/HEAD',
  '/Users/johan/test/test-repos/.git/MERGE_RR',
  '/Users/johan/test/test-repos/.git/ORIG_HEAD',
  '/Users/johan/test/test-repos/.git/config',
  '/Users/johan/test/test-repos/.git/index',
  '/Users/johan/test/test-repos/.git/packed-refs',
  '/Users/johan/test/test-repos/.git/pre-commit',
  '/Users/johan/test/test-repos/.git/smartgit.config',
  '/Users/johan/test/test-repos/node_modules/.yarn-integrity',
  '/Users/johan/test/test-repos/games/.DS_Store',
  '/Users/johan/test/test-repos/games/README.md'
] 30
```

Without the maxDepth option, fdir properly scans through and finds `204516` files

OSX Catalina / APFS

**Comment by thecodrr** [comment]
Hey, nice catch. The `maxDepth` logic was incorrect. Fixed it now. Will be publishing a new version soon. Thanks

**Comment by thecodrr** [comment]
Fixed in v3.0
